cpp/epidemiology/abm/person.cpp

```diff
@@ -115,4 +128,31 @@ bool Person::goes_to_school(TimePoint t, const AbmMigrationParameters& params) c
 {
     return m_random_schoolgroup < params.get<SchoolRatio>().get_matrix_at(t.days())[0];
 }
+
+bool Person::get_tested(const TestParameters& params)
+{
+    double random = UniformDistribution<double>::get_instance()();
+    if (m_state == InfectionState::Carrier || m_state == InfectionState::Infected ||
+        m_state == InfectionState::Infected_Severe || m_state == InfectionState::Infected_Critical) {
+        if (random < params.sensitivity) {
+            m_quarantine = true;
+            return true;
+        }
+        else {
+            m_quarantine               = false;
+            m_time_since_negative_test = days(0);
+            return false;
+        }
+    }
+    else {
+        if (random < params.specificity) {
+            m_quarantine               = false;
+            m_time_since_negative_test = days(0);
+            return false;
+        }
+        else {
+            return true;
+        }
+    }
```

Comment:
maybe I'm misremembering, or may you aren't finished with the changes yet, but didn't we agree to ignore tests of healthy persons until we do re-testing to get them out of quarantine?

Reply:
Oh, I thought we agreed on not putting healthy persons in quarantine (even if they tested positive) but keep testing them. Do you think we should not test them at all/ just give all healthy people a positive test result?

Reply:
I think you mean "a negative test result". that was what I was thinking about. but this solution is fine, we get at least part of the effect of false positives.